### PR TITLE
Add support for Beyondcorp AppConnector resource

### PR DESCRIPTION
**Pull request by palramanathan**
Add terraform support for BeyondCorp AppConnector resource (https://cloud.google.com/beyondcorp/docs/reference/rest/v1/projects.locations.appConnectors) and AppGateway resource(https://cloud.google.com/beyondcorp/docs/reference/rest/v1/projects.locations.appGateways). Adding API and Terraforms yamls and couple of examples.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**
```release-note:new-resource
`google_beyondcorp_app_connector`
```
```release-note:new-resource
`google_beyondcorp_app_gateway`
```

### CODE REVIEW

File: mmv1/products/beyondcorp/api.yaml [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+      data from users over the connection.
+    min_version: ga
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/beyondcorp-enterprise/docs/enable-app-connector'
+      api: 'https://cloud.google.com/beyondcorp/docs/reference/rest#rest-resource:-v1.projects.locations.appconnectors'
+    base_url: projects/{{project}}/locations/{{region}}/appConnectors
+    self_link: projects/{{project}}/locations/{{region}}/appConnectors/{{name}}
+    create_url: projects/{{project}}/locations/{{region}}/appConnectors?app_connector_id={{name}}
+    delete_url: projects/{{project}}/locations/{{region}}/appConnectors/{{name}}
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:43]: delete & update URL should be redundant, they can generally be derived from self_link & base_url as long as they are standard (which these are)

File: mmv1/products/beyondcorp/api.yaml [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+          - true
+          - false
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        required: true
+        input: true
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:72]: If you mark this url_param_only this won't be sent in the request body or read from the response body, removing the need for the custom expand/flatteners. Usually this is what we want as the `name` field is almost always ignored in request bodies, and can't change so it doesn't matter if we read it back

File: mmv1/products/beyondcorp/api.yaml [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        description: |
+          Resource labels to represent user provided metadata.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'principalInfo'
+        description: |
+          Principal information about the Identity of the AppConnector.
+        required: true
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'serviceAccount'
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:96]: Should this be required as well?

File: mmv1/products/beyondcorp/api.yaml [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            description: |
+              ServiceAccount represents a GCP service account.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'email'
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:101]: Same as above, required?

File: mmv1/products/beyondcorp/api.yaml [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+                description: |
+                  Email address of the service account.
+      - !ruby/object:Api::Type::Time 
+        name: 'createTime'
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:105]: Are these output only fields useful? We try to omit them when we don't expect any user to use them, like create & update time

File: mmv1/products/beyondcorp/api.yaml [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+        'Official Documentation': 'https://cloud.google.com/beyondcorp-enterprise/docs/enable-app-connector'
+      api: 'https://cloud.google.com/beyondcorp/docs/reference/rest#rest-resource:-v1.projects.locations.appconnectors'
+    base_url: projects/{{project}}/locations/{{region}}/appConnectors
+    self_link: projects/{{project}}/locations/{{region}}/appConnectors/{{name}}
+    create_url: projects/{{project}}/locations/{{region}}/appConnectors?app_connector_id={{name}}
+    delete_url: projects/{{project}}/locations/{{region}}/appConnectors/{{name}}
+    update_url: projects/{{project}}/locations/{{region}}/appConnectors/{{name}}
+    update_verb: :PATCH
+    update_mask: true
+    # iam_policy: !ruby/object:Api::Resource::IamPolicy
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:47]: You can delete these

File: mmv1/products/beyondcorp/api.yaml [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+          - :UPDATING
+          - :DELETING
+          - :DOWN
+  - !ruby/object:Api::Resource
+    name: 'AppGateway'
+    description: |
+      A BeyondCorp AppGateway resource represents a BeyondCorp protected AppGateway to a remote application. It creates 
+      all the necessary GCP components needed for creating a BeyondCorp protected AppGateway. Multiple connectors can be 
+      authorised for a single AppGateway.
+    min_version: ga
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:137]: Unneeded, GA is the "highest" version

File: mmv1/products/beyondcorp/api.yaml [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/beyondcorp-enterprise/docs/enable-app-connector'
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:140]: Should this link to an app gateways doc?

File: mmv1/products/beyondcorp/api.yaml [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+        allowed:
+          - true
+          - false
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        required: true
```
Comment (slevenick) [inline mmv1/products/beyondcorp/api.yaml:173]: Same as the note on name above

**Review by slevenick — CHANGES_REQUESTED** [review]
Overall looks reasonable, just some notes on cleaning up the api.yaml file

Additionally, you've included info on using PATCH and an update mask to update the resources, can you add a handwritten update test to make sure this works correctly?

**Review by palramanathan — COMMENTED** [review]
Thanks for the review! I will address the comments and update the PR.

**Review by slevenick — APPROVED** [review]
Thanks looks good!

**Pull request merged**